src/olympia/stats/decorators.py

```diff
@@ -0,0 +1,22 @@
+import functools
+
+from olympia.access import acl
+from olympia.addons.models import Addon
+from olympia.amo import permissions
+from olympia.addons.decorators import addon_view
+
+
+def addon_view_stats(f):
+    @functools.wraps(f)
+    def wrapper(request, addon_id=None, *args, **kw):
+        """Provides an addon instance to the view given addon_id, which can be
+        an Addon pk or a slug."""
+        assert addon_id, 'Must provide addon id or slug'
```

Comment:
Do you need this?  If they don't provide an addon id or slug it will just 404, as normal, right?

Reply:
I took this from https://github.com/mozilla/addons-server/blob/17b89599da157c65b2a3f976b8e2786eff9d9422/src/olympia/addons/decorators.py#L23. Would you like me to remove it?

Reply:
Maybe its to stop empty slugs being matched (for deleted addons).  You're calling `addon_view` further down which checks that so you don't need it here as well, anyway.  Just the permission check seems enough.